extensions/msal-node-extensions/src/broker/NativeBrokerPlugin.ts

```diff
@@ -0,0 +1,415 @@
+                    if (wrappedError) {
+                        reject(wrappedError);
+                        return;
+                    }
+                }
+
+                const accountInfoResult = [];
+                result.accounts.forEach((account: Account) => {
+                    accountInfoResult.push(this.generateAccountInfo(account));
+                });
+                resolve(accountInfoResult);
+            };
+
+            try {
+                msalNodeRuntime.DiscoverAccountsAsync(clientId, correlationId, resultCallback);
+            } catch (e) {
+                const wrappedError = this.wrapError(e);
+                if (wrappedError) {
+                    reject(wrappedError);
+                }
+            }
+        });
+    }
+
+    async acquireTokenSilent(request: NativeRequest): Promise<AuthenticationResult> {
+        this.logger.trace("NativeBrokerPlugin - acquireTokenSilent called", request.correlationId);
+        const authParams = this.generateRequestParameters(request);
+        let account: Account;
+        if (request.accountId) {
+            const readAccountResult = await this.readAccountById(request.accountId, request.correlationId);
+            account = readAccountResult.account;
+        }
+
+        return new Promise((resolve: (value: AuthenticationResult) => void, reject) => {
+            const resultCallback = (result: AuthResult) => {
+                try {
+                    result.CheckError();
+                } catch (e) {
+                    const wrappedError = this.wrapError(e);
+                    if (wrappedError) {
+                        reject(wrappedError);
+                        return;
+                    }
+                }
+                const authenticationResult = this.getAuthenticationResult(request, result);
+                resolve(authenticationResult);
+            };
+
+            try {
+                if (account) {
+                    msalNodeRuntime.AcquireTokenSilentlyAsync(authParams, request.correlationId, account, resultCallback);
+                } else {
+                    msalNodeRuntime.SignInSilentlyAsync(authParams, request.correlationId, resultCallback);
+                }
+            } catch (e) {
+                const wrappedError = this.wrapError(e);
+                if (wrappedError) {
+                    reject(wrappedError);
+                }
+            }
+        });
+    }
+
+    async acquireTokenInteractive(request: NativeRequest, windowHandle?: Buffer): Promise<AuthenticationResult> {
+        this.logger.trace("NativeBrokerPlugin - acquireTokenInteractive called", request.correlationId);
+        const authParams = this.generateRequestParameters(request);
+        let account: Account;
+        if (request.accountId) {
+            const readAccountResult = await this.readAccountById(request.accountId, request.correlationId);
+            account = readAccountResult.account;
+        }
```

Comment:
This code is repeated. Should we have a util fn?

Reply:
Done